### PR TITLE
Allow AOs to log in / sign up

### LINKED ISSUE #29: Allow AOs to log in / sign up
When I click to open my data,
    and that I see a page explaining me I'll be redirected to data.gouv.fr,
    I want to understand what's data.gouv.fr,
    and that I want to understand what's the relationship between data.gouv.fr and Transport,
    and that I do not read in english,
    and that I want to know exactly what I'm supposed to do,
    so I can be reassured that I'm doing the right thing,
    and that I wont be left to my fate.

Scénario :

- [x] I can see a log in / sign up link #35
- [x] I have an explanation of what data.gouv.fr is #67 
- [x] I have an explanation of what the relationship is between data.gouv.fr and Transport #67 
- [x] I have an explanation of what's going to happen and what I'm I supposed to do #67 
- [x] The explanations are in french #47 
- [x] I can click somewhere to start the log in / sign up process #40 
- [x] I can click somewhere to ask for help #59 

image uploaded from ios

### PR DESCRIPTION
**Pull request by l-vincent-l**
Fixes #29

### CODE REVIEW

File: lib/transport_web/controllers/auth_controller.ex [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  require Logger
+
+  def index(conn, _params) do
+    Logger.debug fn -> inspect(Datagouvfr.authorize_url!()) end
+    redirect(conn, external: Datagouvfr.authorize_url!())
+  end
+
+  def delete(conn, _params) do
+    conn
+    |> put_flash(:info, "Vous êtes déconnecté")
```
Comment (maukoquiroga) [inline lib/transport_web/controllers/auth_controller.ex]: déconnecté·e

File: lib/transport_web/controllers/auth_controller.ex [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    |> configure_session(drop: true)
+    |> redirect(to: "/")
+  end
+
+  def callback(conn, %{"code" => code}) do
+    client = Datagouvfr.get_token!(code: code)
```
Comment (maukoquiroga) [inline lib/transport_web/controllers/auth_controller.ex]:
```elixir
user = Datagouvfr.get_token!(code: code) |> get_user!
```

File: lib/transport_web/oauth/datagouvfr.ex [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  # Public API
+
+  def client do
+    OAuth2.Client.new([
+      strategy: __MODULE__,
+      client_id: System.get_env("DATAGOUVFR_CLIENT_ID"),
+      client_secret: System.get_env("DATAGOUVFR_CLIENT_SECRET"),
+      site: System.get_env("TRANSPORT_SITE"),
+      redirect_uri: "http://localhost:5000/login/oauth/datagouvfr/callback"
```
Comment (maukoquiroga) [inline lib/transport_web/oauth/datagouvfr.ex]: We should add that to an env as well.

File: lib/transport_web/oauth/datagouvfr.ex [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  def authorize_url! do
+    OAuth2.Client.authorize_url!(client(), %{scope: "default"})
+  end
+
+  # you can pass options to the underlying http library via `opts` parameter
+  def get_token!(params \\ [], headers \\ [] ) do
+    OAuth2.Client.get_token!(client(), params)
+  end
+  
+  def authorize_url(client, params) do
```
Comment (maukoquiroga) [inline lib/transport_web/oauth/datagouvfr.ex]: We do you need to pass client here?

File: lib/transport_web/controllers/auth_controller.ex [hunk 5]
```diff
@@ -0,0 +1,31 @@
+defmodule TransportWeb.AuthController do
+  use TransportWeb, :controller
+  require Logger
+
+  def index(conn, _params) do
```
Comment (maukoquiroga) [inline lib/transport_web/controllers/auth_controller.ex]: That should be `new` maybe?

**Comment by maukoquiroga** [comment]
@l-vincent-l To try I should go to `/auth`?

**Comment by l-vincent-l** [comment]
You should go on `/auth/`

File: lib/transport_web/templates/layout/_navigation.html.eex [hunk 6]
```diff
@@ -1,5 +1,9 @@
 <nav id="navigation" role="navigation">
   <div class="navigation-wrapper">
     <%= img_tag("images/logo.svg", alt: "transport.beta.gouv.fr") |> link(to: "/", class: "logo") %>
+
+    <a style="float:right; color: white; margin: 1em;" href="<%= session_path @conn, :new %>">
```
Comment (maukoquiroga) [inline lib/transport_web/templates/layout/_navigation.html.eex]: Just an example, not to merge like that ! Also, we need to add a previous step where we explain they'll be redirected.

**Comment by maukoquiroga** [comment]
Rebase + fixup

**Comment by maukoquiroga** [comment]
Superseded by #40

**Pull request closed without merging**